libafl_libfuzzer/libafl_libfuzzer_runtime/src/schedulers.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+    collections::{BTreeSet, HashMap},
+    marker::PhantomData,
+};
+
+use libafl::{
+    corpus::{Corpus, CorpusId, Testcase},
+    feedbacks::MapNoveltiesMetadata,
+    inputs::UsesInput,
+    schedulers::{RemovableScheduler, Scheduler},
+    state::{HasCorpus, HasMetadata, UsesState},
+    Error,
+};
+
+#[derive(Clone, Debug)]
+pub struct MergeScheduler<S> {
+    mapping: HashMap<usize, CorpusId>,
+    all: BTreeSet<CorpusId>,
+    phantom: PhantomData<S>,
+}
+
+impl<S> UsesState for MergeScheduler<S>
+where
+    S: UsesInput,
+{
+    type State = S;
+}
+
+impl<S> RemovableScheduler for MergeScheduler<S>
```

Comment:
Same question, shoudl they go in the main lib?

Reply:
In this case, definitely not. This only is relevant for strictly merging fuzzer campaigns.